### PR TITLE
[#5514] Fix DiskFileUpload and MemoryFileUpload equals(...) method.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

DiskFileUpload and MemoryFileUpload.equals(...) are broken.

Modifications:

Fix implementation and add unit test.

Result:

Equals method are correct now.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java [hunk 1]
```diff
@@ -69,16 +69,12 @@ public void setFilename(String filename) {
...
     }
 
     @Override
     public boolean equals(Object o) {
-        if (!(o instanceof Attribute)) {
-            return false;
-        }
-        Attribute attribute = (Attribute) o;
-        return getName().equalsIgnoreCase(attribute.getName());
+        return !(o instanceof FileUpload) ? false : FileUploadUtil.equals(this, (FileUpload) o);
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java]: MAJOR Ternary operators that can be simplified with || or && [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifiedTernary)

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java [hunk 2]
```diff
@@ -63,16 +63,12 @@ public void setFilename(String filename) {
...
     }
 
     @Override
     public boolean equals(Object o) {
-        if (!(o instanceof Attribute)) {
-            return false;
-        }
-        Attribute attribute = (Attribute) o;
-        return getName().equalsIgnoreCase(attribute.getName());
+        return !(o instanceof FileUpload) ? false : FileUploadUtil.equals(this, (FileUpload) o);
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java]: MAJOR Ternary operators that can be simplified with || or && [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifiedTernary)

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return upload1.getName().equalsIgnoreCase(upload2.getName());
+    }
+
+    static int compareTo(FileUpload upload1, FileUpload upload2) {
+        int v;
+        v = upload1.getName().compareToIgnoreCase(upload2.getName());
+        if (v != 0) {
+            return v;
+        }
+        // TODO should we compare size for instance ?
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java]: INFO Complete the task associated to this TODO comment. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1135)

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        return v;
+    }
+
```
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java]: nit: kill line

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java (hunk 3 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java]: do we have any context for how this was intended to be used? It seems strange to compare the length, but not compare the actual content. If we don't do this consider just making this method `return upload1.getName().compareToIgnoreCase(upload2.getName());`

Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/FileUploadUtil.java]: @Scottmitch no context here.. Let me just remove the TODO and change to what you suggested.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (f012d4c) and 4.1 (c735b3e)

**Pull request closed without merging**